app/components/shared/inputs/TimePickerInput.vue.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Prop } from 'vue-property-decorator';
+import VueTimepicker from 'vue2-timepicker';
+import { BaseInput } from './BaseInput';
+import { ITimeMetadata } from './index';
+
+export interface ITimeInputValueMetadata {
+  HH?: string;
+  H?: string;
+  hh?: string;
+  a?: string;
+  A?: string;
+  h?: string;
+  kk?: string;
+  k?: string;
+  m?: string;
+  mm?: string;
+  s?: string;
+  ss?: string;
+}
+
+@Component({
+  components: {
+    VueTimepicker
+  }
+})
+export default class TimePickerInput extends BaseInput<ITimeInputValueMetadata, ITimeMetadata> {
+  @Prop() value: ITimeInputValueMetadata;
```

Comment:
That mixed me up. The value here is more `data` than `metadata`. https://en.wikipedia.org/wiki/Metadata
I think you need something like `ITimeInputValue` or `ITime`